### PR TITLE
feat: add context to the component to avoid props drilling

### PR DESCRIPTION
**Pull request by Mikdaam**

### CODE REVIEW

File: frontend/src/pages/research-configuration/components/DropConfig.jsx [hunk 1]
```diff
@@ -42,4 +42,36 @@ const DropConfig = ({researchId, show, handleClose}) => {
     )
 }
 
-export default DropConfig;
\ No newline at end of file
+export default DropConfig;
```
Comment (P-DAO) [inline frontend/src/pages/research-configuration/components/DropConfig.jsx:45]: a supp

File: frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx [hunk 2]
```diff
@@ -35,7 +37,7 @@ const SamplingAlgorithmForm = ({entityName, algorithms, onConfigChange}) => {
     return (
         <Col>
             <span>{entityName}</span>
-            <SelectSamplingAlgo algorithms={algorithms} onSelect={setSelectedAlgorithmName} />
+            <SelectSamplingAlgo algorithms={listSamplingAlgorithm} onSelect={setSelectedAlgorithmName}/>
```
Comment (P-DAO) [inline frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx:40]:
```suggestion
            <SelectSamplingAlgo algorithms={listSamplingAlgorithm} onSelect={setSelectedAlgorithmName} />
```

File: frontend/src/Router.js [hunk 3]
```diff
@@ -14,7 +15,9 @@ const router = createBrowserRouter([
     },
     {
         path: "/research/:researchId/config",
-        element: <FormSearchConfig />
+        element: <ResearchConfigProvider>
+            <ResearchConfigForm/>
```
Comment (julienleows) [inline frontend/src/Router.js:19]:
```suggestion
            <ResearchConfigForm />
```

File: frontend/src/pages/research-configuration/sampling/SamplingStep.jsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { Container, Row } from 'react-bootstrap';
 
 import SamplingAlgorithmForm from './components/SamplingAlgorithmForm';
 import ResponseError from '../../../ResponseError';
+import { useResearchConfig } from "../ResearchConfigContext";
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/sampling/SamplingStep.jsx:6]:
```suggestion
import { useResearchConfig } from '../ResearchConfigContext';
```

File: frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx (hunk 2 above)
Comment (julienleows) [inline frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx:40]:
```suggestion
            <SelectSamplingAlgo algorithms={listSamplingAlgorithm} onSelect={setSelectedAlgorithmName} />
```

**Review by julienleows — CHANGES_REQUESTED** [review]

**Review by P-DAO — APPROVED** [review]

File: frontend/src/Router.js (hunk 3 above)
Comment (julienleows) [inline frontend/src/Router.js:19]:
```suggestion
            <ResearchConfigForm />
```

File: frontend/src/pages/clerical-review/details/listComparisons/ToCompareList.jsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { useState, useCallback, useEffect } from "react";
-import { ListGroup, Button  } from "react-bootstrap";
+import { useCallback, useEffect, useState } from "react";
+import { Button, ListGroup } from "react-bootstrap";
 import ToCompareItem from "./ToCompareItem";
```
Comment (julienleows) [inline frontend/src/pages/clerical-review/details/listComparisons/ToCompareList.jsx:3]:
```suggestion
import ToCompareItem from './ToCompareItem';
```

File: frontend/src/pages/research-configuration/ResearchConfigForm.jsx [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
+import { useEffect } from 'react';
+import { Button } from 'react-bootstrap';
+import { Stepper } from 'react-form-stepper';
 
 import DropConfig from './components/DropConfig';
 import SamplingStep from './sampling/SamplingStep';
 import FieldSelector from './fields/FieldSelector';
-import Title from './components/Title';
 import DataPreviewStep from "./dataPreview/DataPreviewStep";
+import { useResearchConfig } from "./ResearchConfigContext";
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/ResearchConfigForm.jsx:9]:
```suggestion
import DataPreviewStep from './dataPreview/DataPreviewStep';
import { useResearchConfig } from './ResearchConfigContext';
```

File: frontend/src/pages/research-configuration/ResearchConfigForm.jsx [hunk 7]
```diff
@@ -137,9 +129,8 @@ const FormSearchConfig = () => {
     return (
         <div className="research-config-container">
             <div className="research-config-header">
-                <DropConfig researchId={researchId} show={show} handleClose={handleClose} />
-                <Title leftEntityName={research.leftEntityName} rightEntityName={research.rightEntityName} />
-                <Stepper steps={labels} activeStep={activeStep}/>
+                <DropConfig/>
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/ResearchConfigForm.jsx:132]:
```suggestion
                <DropConfig />
```

File: frontend/src/pages/research-configuration/ResearchConfigForm.jsx [hunk 8]
```diff
@@ -137,9 +129,8 @@ const FormSearchConfig = () => {
...
+                <Stepper steps={steps.map(step => ({label: step.label}))} activeStep={activeStep}/>
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/ResearchConfigForm.jsx:133]:
```suggestion
                <Stepper steps={steps.map(step => ({label: step.label}))} activeStep={activeStep} />
```

File: frontend/src/pages/research-configuration/components/DropConfig.jsx [hunk 9]
```diff
@@ -1,45 +1,44 @@
-import { useCallback } from 'react';
 import Button from 'react-bootstrap/Button';
 import Modal from 'react-bootstrap/Modal';
-import { useNavigate } from 'react-router-dom';
+import { useResearchConfig } from "../ResearchConfigContext";
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/components/DropConfig.jsx:3]:
```suggestion
import { useResearchConfig } from '../ResearchConfigContext';
```

File: frontend/src/pages/research-configuration/sampling/SamplingStep.jsx [hunk 10]
```diff
@@ -36,27 +37,35 @@ const fetchListSamplingAlgorithm = async (setListSamplingAlgorithm, setListSampl
...
         <div>
             <Container>
                 <Row className="mt-3">
                     {listSamplingAlgorithmError && <div>{listSamplingAlgorithmError.message}</div>}
-                    <SamplingAlgorithmForm entityName={"A"} algorithms={listSamplingAlgorithm} onConfigChange={onLeftConfigChange} />
-                    <SamplingAlgorithmForm entityName={"B"} algorithms={listSamplingAlgorithm} onConfigChange={onRightConfigChange} />
+                    <SamplingAlgorithmForm entityName={research.leftEntityName}
+                                           onConfigChange={setLeftSamplerConfiguration}/>
+                    <SamplingAlgorithmForm entityName={research.rightEntityName}
+                                           onConfigChange={setRightSamplerConfiguration}/>
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/sampling/SamplingStep.jsx:64]:
```suggestion
                    <SamplingAlgorithmForm entityName={research.leftEntityName}
                                           onConfigChange={setLeftSamplerConfiguration} />
                    <SamplingAlgorithmForm entityName={research.rightEntityName}
                                           onConfigChange={setRightSamplerConfiguration} />
```

File: frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx [hunk 11]
```diff
@@ -3,6 +3,7 @@ import { Col } from 'react-bootstrap';
 
 import SelectSamplingAlgo from './SelectSamplingAlgo';
 import DynamicForm from './DynamicForm';
+import { useResearchConfig } from "../../ResearchConfigContext";
```
Comment (julienleows) [inline frontend/src/pages/research-configuration/sampling/components/SamplingAlgorithmForm.jsx:6]:
```suggestion
import { useResearchConfig } from '../../ResearchConfigContext';
```

**Review by julienleows — CHANGES_REQUESTED** [review]

**Review by P-DAO — APPROVED** [review]

**Pull request merged**